app/app.go

```diff
@@ -56,40 +59,73 @@ func addGlobalFlags(app *cli.App) {
 			EnvVar: "KORP_GLOBAL_DEBUG",
 		},
 		cli.StringFlag{
-			Name:     "config, c",
+			Name:     "string-config, c",
+			Usage:    "Load configuration from `STRING`",
+			EnvVar:   "KORP_GLOBAL_STRING_CONFIG",
+			FilePath: "./config",
+		},
+		cli.StringFlag{
+			Name:     "config, f",
 			Usage:    "Load configuration from `FILE`",
 			EnvVar:   "KORP_GLOBAL_CONFIG",
 			FilePath: "./config",
 		},
 	}
 }
 
-// TODO to be completed
-// addBefore - Add before-action to CLI application
+// addBefore - Add an action to be execute before the CLI application even starts
 func addBefore(app *cli.App) {
 
 	app.Before = func(c *cli.Context) error {
 
-		if c.Bool("debug") {
-			log.SetLevel(log.DebugLevel)
-		}
-
-		// TODO add config-file action
-		// try using https://github.com/kelseyhightower/envconfig
+		setDebug(c.Bool("debug"))
+		loadConfigAsString(c.String("string-config"))
 
 		return nil
 	}
 }
 
+// setDebug -
+func setDebug(debug bool) {
+
+	if debug {
+		korp_utils.SetLogLevel("debug")
+	}
+}
+
+// loadConfigAsString -
+func loadConfigAsString(wholeConfig string) {
```

Comment:
I don't think this is the way to go. I think we should use the pattern suggested by the framework, using [altsrc](https://kubernetes.io/docs/tasks/administer-cluster/access-cluster-api/#without-kubectl-proxy).

Reply:
I need to have a look at that, anyway this was just a workaround waiting for a more elegant solution.